### PR TITLE
feat(dop,gittar): Support for multi-branch merge

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Support for multi-branch merge

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjUwLDEyMThdLCJhc3NpZ25lZSI6WyIxMDAxMjYxIl19&id=310362&iterationID=1250&pId=311462&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Support for multi-branch merge          |
| 🇨🇳 中文    |      支持多分支合并        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: api/proto/apps/devflow/flow/flow.proto [hunk 1]
```diff
@@ -2,59 +2,65 @@ syntax = "proto3";
...
         option(erda.common.openapi) = {
-            path : "/api/devflow/actions/reconstruction",
+            path: "/api/devflow/actions/reconstruction",
         };
     }
-    rpc GetDevFlowInfo (GetDevFlowInfoRequest) returns (GetDevFlowInfoResponse) {
+    rpc GetDevFlowInfo(GetDevFlowInfoRequest) returns (GetDevFlowInfoResponse) {
         option(google.api.http) = {
-            get : "/api/devflow/actions/info",
+            get: "/api/devflow/actions/info",
```
Comment (sfwn) [inline api/proto/apps/devflow/flow/flow.proto:54]: I think `/api/devflow/{flowID}` is more restful

File: internal/apps/devflow/flow/flow.service.go [hunk 2]
```diff
@@ -364,41 +517,100 @@ func (s *Service) GetDevFlowInfo(ctx context.Context, req *pb.GetDevFlowInfoRequ
 				}()
 				return nil
 			}
+			if _, ok := targetBranchTempBranchMap[mrInfo.TargetBranch]; !ok {
+				rule, err := s.p.DevFlowRule.GetFlowByRule(ctx, devflowrule.GetFlowByRuleRequest{
+					ProjectID:    req.ProjectID,
+					FlowType:     "multi_branch",
```
Comment (sfwn) [inline internal/apps/devflow/flow/flow.service.go:523]: Any type for `FlowType` field?

File: internal/apps/devflow/flow/flow.service.go [hunk 3]
```diff
@@ -414,26 +626,27 @@ func (s *Service) GetDevFlowInfo(ctx context.Context, req *pb.GetDevFlowInfoRequ
 			extra := relationExtraMap[relationID]
 			mrInfo := relationMergeInfoMap[relationID]
 			appDto := appInfoMap[extra.AppID]
+			repoPath := gittarPrefixOpenApi + appDto.ProjectName + "/" + appDto.Name
```
Comment (sfwn) [inline internal/apps/devflow/flow/flow.service.go:629]: you should ensure `gittarPrefixOpenApi` has `/` at the end.

File: internal/tools/gittar/api/merge.go [hunk 4]
```diff
@@ -505,3 +506,79 @@ func GetMergeRequestsStats(ctx *webcontext.Context) {
...
+	commitMessage := fmt.Sprintf("Merge branch '%s' into '%s'", req.SourceBranch, req.TargetBranch)
+	commit, err := ctx.Repository.Merge(req.SourceBranch, req.TargetBranch, ctx.User.ToGitSignature(), commitMessage)
+	if err != nil {
+		ctx.Abort(err)
+		return
+	}
+	ctx.Success(commit)
+}
+
+func MergeBase(ctx *webcontext.Context) {
```
Comment (sfwn) [inline internal/tools/gittar/api/merge.go:562]: GetMergeBase

**Review by sfwn — CHANGES_REQUESTED** [review]

File: api/proto/apps/devflow/flow/flow.proto [hunk 5]
```diff
@@ -2,59 +2,65 @@ syntax = "proto3";
...
         };
         option(erda.common.openapi) = {
-            path : "/api/devflow/actions/info",
+            path: "/api/devflow/actions/info",
+        };
+    }
+    rpc JoinTempBranchByGittarPushHook(GittarPushPayloadEvent)
+            returns (JoinTempBranchByGittarPushHookResponse) {
+        option(google.api.http) = {
+            post: "/api/devflow/actions/actions/join-by-gittar-push-hook",
```
[the remark this answers is not in the document]
Comment (littlejiancc) [reply api/proto/apps/devflow/flow/flow.proto:63]: done

File: internal/tools/gittar/api/merge.go (hunk 4 above)
Comment (littlejiancc) [reply internal/tools/gittar/api/merge.go:562]: done

File: internal/apps/devflow/flow/flow.service.go (hunk 3 above)
Comment (littlejiancc) [reply internal/apps/devflow/flow/flow.service.go:629]: done

File: api/proto/apps/devflow/flow/flow.proto (hunk 1 above)
Comment (littlejiancc) [reply api/proto/apps/devflow/flow/flow.proto:54]: there are no flowID

**Review by sfwn — APPROVED** [review]

**Pull request merged**